src/main/java/uk/gov/hmcts/reform/translate/service/DictionaryService.java

```diff
@@ -38,4 +52,46 @@ public Map<String, String> getDictionaryContents() {
 
         return Collections.emptyMap();
     }
+
+
+    public void putDictionary(final DictionaryRequest dictionaryRequest) {
+
+        val currentUser = securityUtils.getUserInfo();
+        dictionaryRequest.getTranslations().entrySet()
+            .stream()
+            .forEach(englishPhrase -> processEachPhrase(englishPhrase, currentUser));
+    }
+
+    private void processEachPhrase(Map.Entry<String, String> currentPhrase, UserInfo currentUser) {
+        val result = dictionaryRepository.findByEnglishPhrase(currentPhrase.getKey());
+        if (result.isPresent()) {
+            updatePhrase(currentPhrase, currentUser, result.get());
+        } else {
+            createNewPhrase(currentPhrase, currentUser);
+        }
+    }
+
+    private void createNewPhrase(Map.Entry<String, String> currentPhrase, UserInfo currentUser) {
+
+        val newEntity = isCurrentRole(currentUser)
+            ? dictionaryMapper.modelToEntityWithTranslationUploadEntity(currentPhrase, currentUser.getUid())
+            : dictionaryMapper.modelToEntityWithoutTranslationPhrase(currentPhrase);
+        dictionaryRepository.save(newEntity);
+    }
+
+    private void updatePhrase(Map.Entry<String, String> currentPhrase,
+                              UserInfo currentUser, DictionaryEntity dictionaryEntity) {
+        if (isCurrentRole(currentUser)) {
+            val translationUploadEntity = new TranslationUploadEntity();
+            translationUploadEntity.setUserId(currentUser.getUid());
```

Comment:
again instead of having to pass in the `currentUser`, could we just call `SecurityUtils.getUserId()` ?

Reply:
well , that could lead in to overhead as return getUserInfo().getUid() in SecurityUtils end up calling UserInfo userInfo = idamRepository.getUserInfo(getUserToken()); It has been discussed already.   But It gave me a good idea  as I only need the userId I will pass the id only . It has been already discussed . DONE